### ISSUE TITLE
Expose GetPropertyChangedSignal declaratively

### ISSUE THREAD

**Issue by LPGhatguy** [Roblox/roact#28]
Right now, you can connect to the `Changed` event in Roact easily with `Roact.Event.Changed`, but using `GetPropertyChangedSignal(name)` on a specific property is much more performant and is much cleaner.

Roact makes that painful right now -- the only way right now is to connect in `didMount` and disconnect in `willUnmount`, typical for wrappers around APIs like this.

I think we can make Roact support `GetPropertyChangedSignal` as a first-class citizen for events.

**Comment by AmaranthineCodices** [comment]
Proposed API:

```
Roact.Change.[PropName]
```

e.g. for a TextBox:

```lua
Roact.createElement("TextBox", {
    [Roact.Change.Text] = function(rbx)
        print("text changed to "..rbx.Text)
    end,
})
```

**Comment by AmaranthineCodices** [comment]
This can be closed now after #51 - I forgot to mention this issue in that PR!

**Comment by LPGhatguy** [comment]
This is super super handy, thanks!
